File: crates/ui/src/button.rs
//! Button component using Tailwind CSS with semantic color tokens.

use maud::{Markup, html};

/// Button size variants.
#[derive(Debug, Clone, Copy, Default)]
pub enum ButtonSize {
    Small,
    #[default]
    Default,
    Large,
}

/// Button style variants.
#[derive(Debug, Clone, Copy, Default)]
pub enum ButtonVariant {
    #[default]
    Primary,
    Secondary,
    Ghost,
}

/// Button component builder.
///
/// # Examples
///
/// ```
/// use ui::{Button, ButtonVariant, ButtonSize};
///
/// // Primary button
/// let btn = Button::new("Click me")
///     .variant(ButtonVariant::Primary)
///     .size(ButtonSize::Default)
///     .render();
///
/// // Disabled secondary button
/// let btn = Button::new("Submit")
///     .variant(ButtonVariant::Secondary)
///     .disabled(true)
///     .render();
///
/// // Small ghost button
/// let btn = Button::new("Cancel")
///     .variant(ButtonVariant::Ghost)
///     .size(ButtonSize::Small)
///     .render();
/// ```
pub struct Button {
    label: String,
    variant: ButtonVariant,
    size: ButtonSize,
    disabled: bool,
}

impl Button {
    /// Create a new button with the given label.
    pub fn new(label: impl Into<String>) -> Self {
        Self {
            label: label.into(),
            variant: ButtonVariant::default(),
            size: ButtonSize::default(),
            disabled: false,
        }
    }

    /// Set the button variant.
    pub fn variant(mut self, variant: ButtonVariant) -> Self {
        self.variant = variant;
        self
    }

    /// Set the button size.
    pub fn size(mut self, size: ButtonSize) -> Self {
        self.size = size;
        self
    }

    /// Set the button as disabled.
    pub fn disabled(mut self, disabled: bool) -> Self {
        self.disabled = disabled;
        self
    }

    /// Render the button.
    pub fn render(self) -> Markup {
        let base = "inline-flex items-center gap-2 font-mono cursor-pointer transition-colors select-none";

        let size = match self.size {
            ButtonSize::Small => "px-2 py-1 text-xs",
            ButtonSize::Default => "px-4 py-2 text-sm",
            ButtonSize::Large => "px-6 py-3 text-base",
        };

        let variant = match self.variant {
            ButtonVariant::Primary => "bg-primary text-primary-foreground border border-primary hover:opacity-90",
            ButtonVariant::Secondary => "bg-secondary text-secondary-foreground border border-border hover:bg-accent",
            ButtonVariant::Ghost => "bg-transparent text-muted-foreground border border-border hover:bg-accent hover:text-accent-foreground",
        };

        let disabled = if self.disabled {
            "opacity-50 cursor-not-allowed pointer-events-none"
        } else {
            ""
        };

        let class = format!("{base} {size} {variant} {disabled}");

        html! {
            button class=(class) disabled[self.disabled] {
                (self.label)
            }
        }
    }
}

File: apps/autopilot-desktop/src/components/Button.ts
/**
 * Basic Button Component
 */

import { html } from "../effuse/template/html"

export interface ButtonProps {
  label: string
  disabled?: boolean
  variant?: "primary" | "secondary" | "danger"
  size?: "sm" | "md" | "lg"
}

export const Button = (props: ButtonProps) => {
  const { label, disabled = false, variant = "primary", size = "md" } = props

  const baseClasses =
    "inline-flex items-center justify-center border border-border bg-background font-semibold uppercase transition duration-150 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-accent focus-visible:ring-offset-0 disabled:pointer-events-none disabled:opacity-50"

  const variantClasses = {
    primary:
      "border-accent text-accent hover:bg-surface-strong",
    secondary:
      "text-foreground hover:bg-surface-strong",
    danger:
      "border-destructive text-destructive hover:bg-surface-strong",
  }[variant]

  const sizeClasses = {
    sm: "h-7 px-2 text-[10px]",
    md: "h-8 px-3 text-[10px]",
    lg: "h-9 px-4 text-[11px]",
  }[size]

  return html`
    <button 
      class="${baseClasses} ${variantClasses} ${sizeClasses}"
      ${disabled ? "disabled" : ""}
      data-ez="button.click"
    >
      ${label}
    </button>
  `
}
